spec/models/assignment/creator_spec.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Assignment::Creator do
+  subject { Assignment::Creator }
+
+  before(:each) do
+    GitHubClassroom.flipper[:assignment_creator].disable
+  end
+
+  describe "#perform" do
+    let(:organization)  { classroom_org }
+    let(:user)          { classroom_teacher }
+    let(:options) do
+      {
+        title: "Assignment 1",
+        slug: "assignment-1",
+        public_repo: true,
+        students_are_repo_admins: 0,
+        invitations_enabled: 1,
+        template_repos_enabled: true,
+        creator: user,
+        organization: organization
+      }
+    end
+
+    context "success" do
+      it "returns success" do
+        result = subject.perform(options: options)
+        expect(result.success?).to be true
+      end
+
+      it "returns the persisted assignment" do
+        result = subject.perform(options: options)
+        expect(result.assignment).not_to be_nil
+        expect(result.assignment.persisted?).to be true
+      end
+    end
+
+    context "failure" do
+      before(:each) do
+        options[:title] = ""
+      end
+
+      it "returns failure" do
+        result = subject.perform(options: options)
+        expect(result.failed?).to be true
+        expect(result.error).to eql("Validation failed: Your assignment title must be present")
+      end
+
+      it "returns result containing unsaved assignment" do
+        result = subject.perform(options: options)
+        expect(result.assignment.persisted?).to be false
+      end
+    end
+  end
+
+  context "deadlines" do
```

Comment:
All of these deadline tests are the exact same tests [taken from `AssignmentsController`](https://github.com/education/classroom/blob/master/spec/controllers/assignments_controller_spec.rb#L133). The old tests in the controller will be removed once we get rid of the feature flag.